staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go

```diff
@@ -439,13 +447,21 @@ func NewConfigFlags(usePersistentConfig bool) *ConfigFlags {
 		// given 25 groups (our groups + a few custom resources) with one-ish version each, discovery needs to make 50 requests
 		// double it just so we don't end up here again for a while.  This config is only used for discovery.
 		discoveryBurst: 100,
+		CacheTTL:       durationPtr(1 * time.Hour),
 	}
 }
 
 func stringptr(val string) *string {
 	return &val
 }
 
+func durationPtr(duration time.Duration) *time.Duration {
+	if duration == 0 {
+		return nil
+	}
+	return &duration
+}
+
```

Comment:
Could you help me understand why a duration of `0` is a `nil`?
Should it not be possible for whatever use-case that we basically do not want to use the cache at all?
Also lmk if I'm missing something here :)

Reply:
I don't have any strong reason to keep the condition but I used this piece of code from the test. So, will drop the condition.
I think duration `0` will expired cache. 
https://github.com/kubernetes/kubernetes/blob/15fb945348cdd65ef398d7e460da96c6c3ca3026/staging/src/k8s.io/client-go/discovery/cached/disk/cached_discovery.go#L157-L159

`The duration in which the cache should be considered valid. If set to zero, the discovery request will not serve from the discovery cache. Default to 1 hour.` will make sense for docs?
Thanks!!

Reply:
Yeah that sounds reasonable to me! I'd also wait for @eddiezane to take a look at this part once since he might have more context.

Provided we do this, you might also want to consider using `k8s.io/utils/pointer` instead of defining your own function:
https://github.com/kubernetes/kubernetes/blob/2078864d0834b0dd61613be581272be6e5caea5f/vendor/k8s.io/utils/pointer/pointer.go#L249-L251

Reply:
done.